tests/DFE.FindInformationAcademiesTrusts.CypressTests/cypress/e2e/regression/trusts/trust-contact-page.cy.ts

```diff
@@ -1,23 +1,117 @@
-            cy.contains("Source and updates").click();
+        it("Checks that a full non DFE email entered returns the correct error message on a TRM ", () => {
+            trustContactsPage
+                .editTRM("Name", "email@hotmail.co.uk");
+            commonPage
+                .checkErrorPopup('Enter a DfE email address without any spaces')
+        })
+
+        it("Checks that an incorrect email entered returns the correct error message on a TRM ", () => {
+            trustContactsPage
+                .editTRM("Name", "email");
+            commonPage
+                .checkErrorPopup('Enter an email address in the correct format, like name@education.gov.uk');
+        })
+
+        it("Checks that illegal characters entered returns the correct error message on a TRM ", () => {
+            trustContactsPage
+                .editTRM("Name", "@£$$^&");
+            commonPage
+                .checkErrorPopup('Enter an email address in the correct format, like name@education.gov.uk');
+        })
+
+        it("Checks that whitespace entered returns the correct error message on a TRM ", () => {
+            trustContactsPage
+                .editTRM("Name", "a     b");
+            commonPage
+                .checkErrorPopup('Enter a DfE email address without any spaces')
+                .checkErrorPopup('Enter an email address in the correct format, like name@education.gov.uk');
+        })
+
+        it("Checks that a full non DFE email entered returns the correct error message on a SFSO ", () => {
+            trustContactsPage
+                .editSFSO("Name", "email@hotmail.co.uk");
+
+            commonPage
+                .checkErrorPopup('Enter a DfE email address without any spaces')
+        })
+
+        it("Checks that an incorrect email entered returns the correct error message on a SFSO ", () => {
+            trustContactsPage
+                .editSFSO("Name", "email");
+            commonPage
+                .checkErrorPopup('Enter a DfE email address without any spaces')
+                .checkErrorPopup('Enter an email address in the correct format, like name@education.gov.uk');
+        })
+
+        it("Checks that illegal characters entered returns the correct error message on a SFSO ", () => {
+            trustContactsPage
+                .editSFSO("Name", "@£$$^&");
+            commonPage
+                .checkErrorPopup('Enter a DfE email address without any spaces')
+                .checkErrorPopup('Enter an email address in the correct format, like name@education.gov.uk');
+        })
 
+        it("Checks that whitespace entered returns the correct error message on a SFSO ", () => {
+            trustContactsPage
+                .editSFSO("Name", "a     b");
+            commonPage
+                .checkErrorPopup('Enter a DfE email address without any spaces')
+                .checkErrorPopup('Enter an email address in the correct format, like name@education.gov.uk');
         })
```

Comment:
I'd probably like to see one more test where the email string is "@education.gov.uk" and the Error message should only say "'Enter an email address in the correct format, like name@education.gov.uk"

Also we might want to do negative tests for ones where only one message is being displayed to check no other messages are present.

Reply:
Now rolled both into the latest push